cura/UI/WelcomePagesModel.py

```diff
@@ -0,0 +1,75 @@
+                            "page_url": QUrl.fromLocalFile(Resources.getPath(CuraApplication.ResourceTypes.QmlFiles,
+                                                                             os.path.join("WelcomePages",
+                                                                                          "WelcomeContent.qml"))),
+                            })
+        self._pages.append({"id": "user_agreement",
+                            "page_url": QUrl.fromLocalFile(Resources.getPath(CuraApplication.ResourceTypes.QmlFiles,
+                                                                             os.path.join("WelcomePages",
+                                                                                          "UserAgreementContent.qml"))),
+                            })
+        self._pages.append({"id": "whats_new",
+                            "page_url": QUrl.fromLocalFile(Resources.getPath(CuraApplication.ResourceTypes.QmlFiles,
+                                                                             os.path.join("WelcomePages",
+                                                                                          "WhatsNewContent.qml"))),
+                            })
+        self._pages.append({"id": "data_collections",
+                            "page_url": QUrl.fromLocalFile(Resources.getPath(CuraApplication.ResourceTypes.QmlFiles,
+                                                                             os.path.join("WelcomePages",
+                                                                                          "DataCollectionsContent.qml"))),
+                            })
+        self._pages.append({"id": "add_printer_by_selection",
+                            "page_url": QUrl.fromLocalFile(Resources.getPath(CuraApplication.ResourceTypes.QmlFiles,
+                                                                             os.path.join("WelcomePages",
+                                                                                          "AddPrinterBySelectionContent.qml"))),
+                            })
+        self._pages.append({"id": "add_printer_by_ip",
+                            "page_url": QUrl.fromLocalFile(Resources.getPath(CuraApplication.ResourceTypes.QmlFiles,
+                                                                             os.path.join("WelcomePages",
+                                                                                          "AddPrinterByIpContent.qml"))),
+                            })
+        self._pages.append({"id": "cloud",
+                            "page_url": QUrl.fromLocalFile(Resources.getPath(CuraApplication.ResourceTypes.QmlFiles,
+                                                                             os.path.join("WelcomePages",
+                                                                                          "CloudContent.qml"))),
+                            })
```

Comment:
Only append these if the plug-in is enabled.

Reply:
Simplified this code. I discussed with @rburema on this and we think it still makes sense to put the cloud page here due to promotion reasons, plus the Cloud plugin is also bundled.